lib/streamlit/report_session.py

```diff
@@ -171,6 +175,15 @@ def enqueue(self, msg):
             if scriptrunner is not None:
                 scriptrunner.maybe_handle_execution_control_request()
 
+        if msg.HasField("page_config_changed") and not self._set_page_config_allowed:
```

Comment:
To be honest, I'm not super familiar with the structore of ReportContext vs ReportThread vs ReportSession; let me know if another location would be more appropriate for this logic.